sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/data/ImmutableStatus.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.StatusCanonicalCode;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Defines the status of a {@link Span} by providing a standard {@link StatusCanonicalCode} in
+ * conjunction with an optional descriptive message. Instances of {@code Status} are created by
+ * starting with the template for the appropriate {@link StatusCanonicalCode} and supplementing it
+ * with additional information: {@code Status.NOT_FOUND.withDescription("Could not find
+ * 'important_file.txt'");}
+ */
+@AutoValue
+@Immutable
+public abstract class ImmutableStatus implements SpanData.Status {
```

Comment:
Since we are on Java 8 now we shouldn't have to expose this class. Can you add `ok()`, etc static methods on the interface which return the values here that can be made package private?

Reply:
We may be able to do that, but I would do the same for ImmutableLink and ImmutableEvent then. Filed https://github.com/open-telemetry/opentelemetry-java/issues/1745 to track this.